duckdb-vortex/src/rust_vector_buffer.cpp

```diff
@@ -0,0 +1,17 @@
+#include "rust_vector_buffer.hpp"
+#include "duckdb/common/types/vector_buffer.hpp"
+#include "duckdb/common/types/vector.hpp"
+
+extern "C" {
+CppVectorBuffer NewCppVectorBuffer(FFIDuckDBBuffer *buffer) {
+	auto rbuffer = duckdb::make_shared_ptr<RustVectorBuffer>(buffer);
+	auto ribuffer = new _CppVectorBufferInternal {.buffer = rbuffer};
+	return reinterpret_cast<CppVectorBuffer>(ribuffer);
+}
+
+void AssignBufferToVec(duckdb_vector vec, CppVectorBuffer *buffer) {
```

Comment:
As `CppVectorBuffer` is typedefd to a pointer, we're passing a pointer to a pointer here.